### PR TITLE
Start a doc for well-known labels & taints

### PR DESCRIPTION
**Pull request by justinsb**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33752)

### CODE REVIEW

**Comment by justinsb** [comment]
Starting an allocation-mutex doc, which I realized we don't have but need for things like https://github.com/kubernetes-incubator/spartakus/issues/4 and https://github.com/kubernetes/kubernetes/issues/29738

File: docs/api-reference/labels-annotations-taints.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+Kubelet populates this with `runtime.GOOS` as defined by Go.  This can be handy if you are mixing operating systems
+in your cluster (although currently Linux is the only OS supported by kubernetes).
+
+## kubernetes.io/hostname
+
+Example: `kubernetes.io/hostname=ip-172-20-114-199.ec2.internal`
+
+Used on: Node
+
+Kubelet populates this with the hostname.  Note that the hostname can be changed from the "actual" hostname
```
Comment (thockin) [inline docs/api-reference/labels-annotations-taints.md]: Can we say what assumptions can be made about hostnames?  Are they DNS resolvable? I don't actually know.

File: docs/api-reference/labels-annotations-taints.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+by passing the `--hostname-override` flag to kubelet.
+
+## beta.kubernetes.io/instance-type
+
+Example: `beta.kubernetes.io/instance-type=m3.medium`
+
+Used on: Node
+
+Kubelet populates this with the instance type as defined by the `cloudprovider`.  It will not be set if
```
Comment (thockin) [inline docs/api-reference/labels-annotations-taints.md]: @wlan0 - one for cloud provider extraction.  This should probably move to a controller in the cloud controller.

File: docs/api-reference/labels-annotations-taints.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+Example:
+
+`failure-domain.beta.kubernetes.io/region=us-east-1`
+
+`failure-domain.beta.kubernetes.io/zone=us-east-1c`
+
+Used on: Node, PersistentVolume
+
+On the Node: Kubelet populates this with the zone information as defined by the `cloudprovider`.  It will not be set if
```
Comment (thockin) [inline docs/api-reference/labels-annotations-taints.md]: @wlan0

**Comment by thockin** [comment]
Good start - now to spread the word far and wide that this is a) required and b) we need help back-populating it.

**Review by thockin — APPROVED** [review]

**Pull request merged**